MobileDevProject/ViewController.swift

```diff
@@ -10,32 +10,37 @@ import CoreLocation
 import UIKit
 
 class ViewController: UIViewController, CLLocationManagerDelegate {
-
+    
     @IBOutlet weak var beaconButton1: UIButton!
     @IBOutlet weak var beaconButton2: UIButton!
     @IBOutlet weak var beaconButton3: UIButton!
-
+    
     @IBOutlet weak var mapRotatingSwitch: UISwitch!
-
+    
     @IBOutlet weak var mapView: UIView!
     @IBOutlet weak var compassImage: UIImageView!
-
+    
     var beaconInfo: [BeaconInfo] = []
+    var beaconsArray: [CLBeacon] = []
+    
+    var nearestBeacon: BeaconInfo!
+    var navigatingBeacon: BeaconInfo! // this is the beacon that the user chose to navigate to
+    var isNavigating: Bool = false
+    
     var mapHeading: Double = 0
     var compassHeading: Double = 0
     var mapIsRotating: Bool = false
+    
     var locationManager: CLLocationManager!
-    var nearestBeacon: BeaconInfo!
+    
     var lineShapeLayer: CAShapeLayer!
     var circleShapeLayer: CAShapeLayer!
-    var beaconsArray: [CLBeacon] = []
     var circleShapeDrawn: Bool = false
-    var beacon1: CLBeacon!
-    var beacon2: CLBeacon!
-    var cgPoint: CGPoint!
-
-    var isNavigating: Bool = false
-    var navigatingBeacon: BeaconInfo!
+    var beacon1: CLBeacon!  // these three variables are all used by calcXY
+    var beacon2: CLBeacon!  // stored globally to allow them to persist between calls
+    var cgPoint: CGPoint!   // this means we can better handle loss of signal
```

Comment:
I don't really know how to make it any clearer than it is already, that can be down to @lexvdherik